protocols/identify/src/identify.rs

```diff
@@ -758,4 +775,29 @@ mod tests {
 
         assert_eq!(connected_peer, swarm1_peer_id);
     }
+
+    #[test]
+    fn check_multiaddr_matches_peer_id() {
+        let peer_id = PeerId::random();
+        let other_peer_id = PeerId::random();
+        let mut addr: Multiaddr = ("/ip4/147.75.69.143/tcp/4001".to_string())
+            .parse()
+            .expect("failed to parse multiaddr");
+
+        let addr_without_peer_id: Multiaddr = addr.clone();
+        let mut addr_with_other_peer_id = addr.clone();
+
+        addr.push(Protocol::P2p(peer_id.clone().into()));
+        addr_with_other_peer_id.push(Protocol::P2p(other_peer_id.into()));
+
+        assert_eq!(true, multiaddr_matches_peer_id(&addr, &peer_id));
```

Comment:
```suggestion
        assert!(multiaddr_matches_peer_id(&addr, &peer_id));
```

Small nit pick. Same for the below.